code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -400,24 +400,32 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 
 		if(agent_b && temp > 500 && air.toxins > MINIMUM_REACT_QUANTITY )
 			var/datum/gas/oxygen_agent_b/trace_gas = payload.get_or_add_trace_gas_by_type(/datum/gas/oxygen_agent_b)
-			payload.temperature = T0C // Greatly reduce temperature to simulate an endothermic reaction
-
-			// Itr 1: 0.125 Agent B, 10 Oxy
+			if(temp>10000)
+				payload.temperature = temp -= 10000 // Greatly reduce temperature to simulate an endothermic reaction
+			else
+				payload.temperature = T0C
+			// Itr 1: 0.2 Agent B, 1 oxy
 			// Itr 2: 0.0605 Agent B
-			// 0.1855moles/12cells=0.0155moles per cell
-			// At 0.0155 moles per cell it will take 20 iterations for reaction rate to drop below MINIMUM_REACT_QUANTITY
-			// Providing 1.3 minutes of catalyst assuming 4 sec ATMOS for a 12 cell burn chamber
-			trace_gas.moles += min(total_oxygen/1024,0.125)
-			total_oxygen -= min(trace_gas.moles*1024,total_oxygen)
-			animate_flash_color_fill_inherit(location,"#FF0000",4, 2 SECONDS)
+			// Should be 100 iterations to deplete total of 19.099 mols of agent B and 100 oxygen, will take 110 iterations to hit minimum reaction rate whihch is about 7.33 minutes, (this is azruns math not mine dont blame me if wrong)
+
+			animate_flash_color_fill_inherit(location,"#ff0000",4, 2 SECONDS)
+			if(!ON_COOLDOWN(location, "sound_cooldownB", 2 SECONDS)) // Prevents ear spam
+				playsound(location, "sound/effects/leakagentb.ogg", 50, 1, 8)
+			if(!particleMaster.CheckSystemExists(/datum/particleSystem/sparklesagentb, location))
+				particleMaster.SpawnSystem(new /datum/particleSystem/sparklesagentb(location))
```

Comment:
Can you add an demo/example of this to the PR?

Reply:
Done! (Added audio showoff too because thats important)